core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1397,20 +1397,15 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     trace("Sending topic metadata %s and brokers %s for correlation id %d to client %s".format(completeTopicMetadata.mkString(","),
       brokers.mkString(","), request.header.correlationId, request.header.clientId))
-    val controllerId = {
```

Comment:
I don't think MetadataResponse should be modified. KIP-919 does not specify any changes to the ControllerId field in this response.

Reply:
Yeah, I know that KIP-919 doesn't mention anything about controller ID in the MetadataResponse. I was just wondering if it makes sense to return a random broker ID as a controller ID, I think it's better to return the real controller ID because now clients are able to directly with controllers.

Reply:
@mumrah is right here. The "real controller ID" is not a broker ID. For example, you might have brokers 0, 1, and 2, and controllers 1001, 1002, and 1003. Returning 1001 would break many clients because they expect to use the returned value as a broker ID. This is explained in various KIPs.